v2/pkg/cli/executor.go

```diff
@@ -323,4 +325,65 @@ func mergeImages(base, in []v1alpha3.CopyImageSchema) []v1alpha3.CopyImageSchema
 func cleanUp() {
 	// clean up logs directory
 	os.RemoveAll(logsDir)
+
+}
+
+func (o *ExecutorSchema) prepareStorage() error {
+	configYamlV0_1 := `
+version: 0.1
+log:
+  fields:
+    service: registry
+storage:
+  cache:
+    blobdescriptor: inmemory
+  filesystem:
+    rootdirectory: $$PLACEHOLDER$$
+http:
+  addr: :5000
+  headers:
+    X-Content-Type-Options: [nosniff]
+      #auth:
+      #htpasswd:
+      #realm: basic-realm
+      #path: /etc/registry
+health:
+  storagedriver:
+    enabled: true
+    interval: 10s
+    threshold: 3
+`
+
+	rootDir := ""
+
+	if o.Opts.Mode == mirrorToDisk {
+		rootDir = strings.TrimPrefix(o.Opts.Destination, fileProtocol)
+	} else {
+		rootDir = strings.TrimPrefix(o.Opts.Global.From, fileProtocol)
+	}
+
+	if rootDir == "" {
+		// something went wrong
+		return fmt.Errorf("error determining the local storage folder to use")
+	}
+	configYamlV0_1 = strings.Replace(configYamlV0_1, "$$PLACEHOLDER$$", rootDir, 1)
+	config, err := configuration.Parse(bytes.NewReader([]byte(configYamlV0_1)))
+
+	if err != nil {
+		return fmt.Errorf("error parsing local storage configuration : %v\n %s\n", err, configYamlV0_1)
+	}
+	fmt.Printf("%v\n", config)
+
+	ctx := dcontext.WithVersion(dcontext.Background(), distversion.Version)
+	reg, err := registry.NewRegistry(ctx, config)
+	if err != nil {
+		return err
+	}
+	o.LocalStorage = *reg
+	go reg.ListenAndServe()
```

Comment:
@sherine-k  what if reg.ListenAndServer() fails ?

Reply:
Didn't Alex add code that uses an error channel to check ?
Of use something similar to this

```
ctx := context.Background()
g, ctx := errgroup.WithContext(ctx)

g.Go(func() error {
	return reg.ListenAndServer()
})
e := g.Wait()

```